flink-runtime/src/main/scala/org/apache/flink/runtime/jobmanager/JobManager.scala

```diff
@@ -1487,7 +1487,7 @@ class JobManager(
           }
         }
 
-        eg.fail(cause)
+        eg.cancel()
```

Comment:
It is important to make sure that HA recovery continues. The `cancelAndClearEverything()` is also triggered in `postStop()` if the actor or actor system crashes (one of JobManager failure scenarios, the process reaper kills the process then).

We hence need a sort of fail that is not locally retried, but does not lead to a terminal state where we run the risk that some cleanup code removes the job from ZooKeeper.
When the job is canceled, don't we run the risk that some code removes the ZooKeeper entries and prohibits HA recovery?